### ISSUE TITLE
Rename return_dict to as_dict in brick applications, and document feature

### ISSUE THREAD

**Issue by bartvm** [bartvm/blocks#208]
Something @pbrakel brought up: Since many applications have an `outputs` attribute, it would be quite easy (and nice) to add a way of accessing the outputs of a brick by the names given in this attribute. For example

```python
@application(inputs=['x'], outputs=['y', 'z'])
def apply(x)
    return (x + 1, x - 1)
```
```python
z = brick.apply(x, as_dict=True)['z']
```

**Comment by rizar** [comment]
Isn't that you need: https://github.com/bartvm/blocks/blob/master/blocks/bricks/base.py#L239?

**Comment by bartvm** [comment]
Ah, so it is there! I thought it was as_dict for some reason and was
surprised it didn't show up when I grepped. We should probably use the same
keyword throughout the library.

**Comment by rizar** [comment]
Don't mind renaming it to "as_dict", 4 letters less :)

**Comment by bartvm** [comment]
Fixed in #359
